_pytest/logging.py

```diff
@@ -376,7 +432,10 @@ def _setup_cli_logging(self):
             log_cli_handler = _LiveLoggingStreamHandler(terminal_reporter, capture_manager)
             log_cli_format = get_option_ini(self._config, 'log_cli_format', 'log_format')
             log_cli_date_format = get_option_ini(self._config, 'log_cli_date_format', 'log_date_format')
-            log_cli_formatter = logging.Formatter(log_cli_format, datefmt=log_cli_date_format)
+            if ColoredLevelFormatter.LEVELNAME_FMT_REGEX.search(log_cli_format):
+                log_cli_formatter = ColoredLevelFormatter(self._config, log_cli_format, datefmt=log_cli_date_format)
```

Comment:
I could also perform the color check in the condition in line 435. Then I don't need to pass `self._config` to the `__init__` method of `ColoredLevelFormatter`.

Reply:
Seems better

Reply:
I didn't know that color is a ternary type ('yes', 'no' and 'auto'). So I can't add `self._config.option.color == 'yes' and ` to the if condition. If the value of the color option is 'auto' the TerminalWriter.__init__ determines if the terminal supports markup or not.

An alternative would be to pass a terminalwriter object (`_pytest.config.create_terminal_writer(self._config)`) as the first arg of ColoredLevelFormatter. Don't know if this makes it better.

Reply:
> An alternative would be to pass a terminalwriter object (_pytest.config.create_terminal_writer(self._config)) as the first arg of ColoredLevelFormatter. Don't know if this makes it better.

This reduces the number of dependencies at least (`ColoredLevelFormatter` will then depend only on `TerminalWriter`, instead of `Config` and `TerminalWriter` with the current code), so feel free to go with this route.